build/src/VersionParser.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace DrupalComposer\DrupalSecurityAdvisories;
+
+class VersionParser {
+
+  public static function generateRangeConstraint($version, $isCore) {
+    if (!static::isValid($version)) {
+      return FALSE;
+    }
+    return $isCore ? static::handleCore($version) : static::handleContrib($version);
+  }
+
+  public static function generateExplicitConstraint($version, $isCore) {
+    if (!static::isValid($version)) {
+      return FALSE;
+    }
+    if ($isCore) {
+      return $version;
+    }
+    else {
+      list($core, $version) = explode('-', $version, 2);
+    }
+    return $version;
+  }
+
+  public static function handleCore($version) {
+    list($major, $minor) = explode('.', $version);
+    return ">=$major.$minor,<$version";
```

Comment:
For D7,  this returns  constraints like `>7.67,<=7.67` instead of `>7,<=7.67`, so a different return is needed

```
  public static function handleCore($version) {
    list($major, $minor) = explode('.', $version);
    if ($major == '7') {
      return ">=$major,<$version";
    }
    return ">=$major.$minor,<$version";
  }
```